drivers/interrupt_controller/intc_arcv2_irq_unit.c

```diff
@@ -170,16 +167,16 @@ static int arc_v2_irq_unit_resume(struct device *dev)
 #endif
 	z_arc_v2_aux_reg_write(_ARC_V2_IRQ_VECT_BASE, ctx.irq_vect_base);
 
-	status32 = z_arc_v2_aux_reg_read(_ARC_V2_STATUS32);
```

Comment:
I think removing the status register 'restore' code here is ok (restored elseware, I assume?), but I suggest to put this in a separate PR as it's not related to irq lock/unlock.